app/services/post_alerter.rb

```diff
@@ -393,18 +393,9 @@ def create_notification(user, type, post, opts = {})
     return if user.staged? && topic.category&.mailinglist_mirror?
 
     notifier_id = opts[:user_id] || post.user_id # xxxxx look at revision history
-
-    # apply muting here
-    return if notifier_id && MutedUser.where(user_id: user.id, muted_user_id: notifier_id)
-      .joins(:muted_user)
-      .where('NOT admin AND NOT moderator')
-      .exists?
-
-    # apply ignored here
-    return if notifier_id && IgnoredUser.where(user_id: user.id, ignored_user_id: notifier_id)
-      .joins(:ignored_user)
-      .where('NOT admin AND NOT moderator')
-      .exists?
+    return if notifier_id && UserCommScreener.new(
+      notifier_id, target_usernames: user.username
+    ).ignoring_or_muting_actor?(user.id)
```

Comment:
I wonder if this can just be a class method: `UserCommScreener.ignoring_or_muting_actor?(notifier_id: notifier_id, target_id: user.id)`. It doesn't seem like there is a need for us to initialise an instance here.

Reply:
I _could_ add class method versions of these but it seems repetitive, since it is valid to initialize `UserCommScreener` and use it for multiple checks, the point is that it does a single query for all the preferences based on the target usernames, I don't think we want to encourage one-shot uses for this.